### DISCUSSION TITLE
time is correlated with the two variables, which we want to assess the dynamics - best practices

### DISCUSSION THREAD

**Discussion by camontal** [cdriveraus/ctsem#41] [General]
Hello,
 
First, a big thank for the amazing package. I am using these models to estimate the dynamics that our data is suggesting, between animal movements and health variables (though we have 3-9 measurements of both variables for each ~400 animals).
 
I am worried about the effect of age because it is correlated with both our movement and health variables. After omitting the first few measurements to reduce the correlation, the pearson correlation of the “time” variable (used in the model) with health is 0.37 and with movement is -0.22 (spearman: 0.43 and -0.21). I tried two approach, which lead to different results:
1.	Using the variable ’time’ as when the measurement took place in months from the first (in term of the smallest animal age) measurement started. In that case because some individuals had their first measurement later than other (~3 weeks diff at max), not all animals have a measure at time = 0.
2.	Using the variable ’time’ as when the measurement took place (in months from the beginning of the study) but starting for each individual at 0. Because we here assume that the age of the first measurement is not of importance, I was hoping to reduce the cofounding effect of time, as the “time” value for each individual represent different age. the pearson correlation of time with health is now 0.36 and time with movement is -0.16 (spearman: 0.39 and -0.16).
 
The directionality of the effect of the health on the movement is the same and seems stable when tested with different random seed, transformation, and subset. However, the effect of movement on health is different: in case 1 it is negative (as we could expect if the age cofounder is present) and in case 2 it is not significant but tend to be positive. I would think that in the first scenario, the time is more of an issue as the second one. I was wondering if you think the second approach sounds oke, since the movement is not correlated to time (<0.20), or if you think I should still be worry about the effect of age.  I thank you very much for any possible insights for using other, better techniques to account for the age.

**Comment by cdriveraus** [comment]
Hi, I think you are right to be worried about age. I don't think either of your approaches really resolves it though, but you're thinking along the right lines. The important thing will be to account for the effect of age, either by including some kind of trend in the model, or by detrending the data in advance. I prefer the former approach as it is a bit more flexible / statistically appropriate, but it is also more complex and offers more scope for screwing things up! To model a separate trend (possibly with random variation over individuals), *around* which you then estimate the dynamics, you basically need to extend the number of latent processes in the model, and have one of them account for the trend (without system noise) and the other account for dynamics / fluctuations. This is described in more detail in[ https://osf.io/gw54v](https://osf.io/gw54v) .

Regarding your two approaches, once the trend is accounted for I think either can work, though you will get a better / simpler trend estimate if you align the times, so that time really reflects age. The best way to do this is probably to create an empty observation row for every individual at the age of your earliest observation -- so if you have one hen observed at 3 days, create an observation at 3 days for all the other hens and just set observed variables to NA.

**Reply by camontal** [reply]
Thank you so much for the quick and very helpful answer! I will try to model a separate trend and add observations as you suggested, thank you!

**Reply by camontal** [reply]
@cdriveraus is there any way to have access to the supplement, or just the model specification from Table A2? I am not always sure my implementation fits what is explained, and would be good to double check if this is a possibility. Again thank you for the help!

**Reply by cdriveraus** [reply]
[supplement.zip](https://github.com/cdriveraus/ctsem/files/10684387/supplement.zip)
Here's the supplement!
